### DISCUSSION TITLE
Ideas for rate limiting users

### DISCUSSION THREAD

**Discussion by brewhousedigital** [pocketbase/pocketbase#3016] [Q&A]
I'm sure this question has been asked somewhere before but I didn't see it in the discussions list. This reddit post for Supabase got me thinking again about rate limits for APIs for Pocketbase. Wondering if anyone has suggestions on the best way to handle this. Middleware? Built into Pocketbase in the future? Has anyone successfully implemented this on their own application?

https://www.reddit.com/r/Supabase/comments/15chrqx/lack_of_rate_limiting_makes_supabase_unsuitable/

**Comment by ganigeorgiev** [comment]
You could subscribe to https://github.com/pocketbase/pocketbase/issues/1232, but for now still remains a low priority.

If you need more advanced network configurations (ip whitelist/blacklist, rate-limitting, caching, etc.) for now you can do that on reverse proxy level (nginx, apache, caddy, etc.).

Rate limiting is not a silver bullet. For paid services like Firebase/Supabase it makes sense to always have it because it can directly affect your usage costs, but when self hosting PocketBase that's usually not the case. Rate limiting could eventually help if you have large/complex queries or some heavy computation task.

But if you are trying to protect your server from DDoS, rate limiting the API will not save you (and depending how it is implemented, it can actually has negative impact on the performance).
If someone wants to flood your server they can just send requests to any endpoint and it doesn't matter whether you'll return 200 or some error code because the request already would have reached the server.

If you want a DDoS protection check your hosting provider whether it doesn't have one by default (Hetzner for example [has it by default for all services](https://www.hetzner.com/unternehmen/ddos-schutz)). If it doesn't, then you need to implement a loadbalancer or some other mechanism that will cut the flood traffic before even reaching your PocketBase server.
